### PR TITLE
DON'T MERGE THIS (adds solution)

### PR DESCRIPTION
**Pull request by tomalley104**
note: uses `willTransitionToTraitCollection` as per current README. **will have to be updated for using `willTransitionToSize` if we refactor the lab.**

### CODE REVIEW

**Comment by TOMalley104** [comment]
also please close issue #1 if you find this solution sufficient

File: horizontal-profile/FISViewController.m [hunk 1]
```diff
@@ -0,0 +1,195 @@
...
+@property (strong, nonatomic) NSLayoutConstraint *coverImageHeightLandscape;
+
+@property (strong, nonatomic) NSArray *textView6PlusLandscape;
+@property (strong, nonatomic) NSArray *textView6PlusPortrait;
+
+@end
+
+@implementation FISViewController
+
+- (void)viewDidLoad
```
Comment (jmburges) [inline horizontal-profile/FISViewController.m]: such a nice `viewDidLoad`

File: horizontal-profile/FISViewController.m [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+    
+    // setup images
+    [self.coverImage setImage: [UIImage imageNamed:@"stuffmomnevertoldyou.jpg"]];
+    [self.profileImage setImage: [UIImage imageNamed:@"Blake_Farenthold.jpg"]];
+    
+    // set constraints
+    [self setPortraitConstraints];
+}
+
+- (void)didReceiveMemoryWarning
```
Comment (jmburges) [inline horizontal-profile/FISViewController.m]: Remove these skeleton methods if you aren't doing anything with them

File: horizontal-profile/FISViewController.m [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+}
+
+-(void)viewDidLayoutSubviews
+{
+    [super viewDidLayoutSubviews];
+}
+
+-(void)willTransitionToTraitCollection:(UITraitCollection *)newCollection withTransitionCoordinator:(id<UIViewControllerTransitionCoordinator>)coordinator
+{
+    //         if UITraitNameVerticalSizeClass = Compact >>> LANDSCAPE
```
Comment (jmburges) [inline horizontal-profile/FISViewController.m]: reason for this comment?

Comment (TOMalley104) [inline horizontal-profile/FISViewController.m]: its supposed to be a chart explaining what the `UITraitName____SizeClass` stuff equates to

File: horizontal-profile/FISViewController.m [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+    }
+    NSLog(@"transitioning to: %@", newCollection);
+    
+    [self.view layoutSubviews];
+}
+
+
+-(void) setCoverImageConstraints
+{
+    NSLayoutConstraint *pinToTop = [NSLayoutConstraint constraintWithItem:self.coverImage attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeTop multiplier:1.0 constant:0];
```
Comment (jmburges) [inline horizontal-profile/FISViewController.m]: so crazy nit-picky....can you multi line these guys? makes it so hard to read

File: horizontal-profile/FISViewController.m [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+    
+    NSLayoutConstraint *setWidth = [NSLayoutConstraint constraintWithItem:self.coverImage attribute:NSLayoutAttributeWidth relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeWidth multiplier:1.0 constant:0];
+    
+    NSLayoutConstraint *setHeight = [NSLayoutConstraint constraintWithItem:self.coverImage attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeHeight multiplier:.25 constant:0];
+    
+    NSLayoutConstraint *setCenter = [NSLayoutConstraint constraintWithItem:self.coverImage attribute:NSLayoutAttributeCenterX relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeCenterX multiplier:1 constant:0];
+    
+    // save these guys for later 
+
+    NSLayoutConstraint *setCIHeightLandscape = [NSLayoutConstraint constraintWithItem:self.coverImage attribute:NSLayoutAttributeHeight relatedBy:NSLayoutRelationEqual toItem:self.view attribute:NSLayoutAttributeHeight multiplier:0 constant:0];
```
Comment (jmburges) [inline horizontal-profile/FISViewController.m]: What's CI in `setCIHeightLandscape`?

Comment (jmburges) [inline horizontal-profile/FISViewController.m]: cover image. get it. bad variable name

File: horizontal-profile/FISViewController.m (hunk 4 above)
Comment (TOMalley104) [inline horizontal-profile/FISViewController.m]: no prob. 

wanna share that plug-in that you used to use in class for that? would make my life easier :)

File: horizontal-profile/FISViewController.m (hunk 5 above)
Comment (TOMalley104) [inline horizontal-profile/FISViewController.m]: agreed and noted

File: horizontal-profile/FISViewController.m [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+    self.coverImageHeightLandscape = setCIHeightLandscape;
+    self.coverImageHeightPortrait = setHeight;
+    
+    [self.view addConstraints:@[pinToTop, setWidth, setHeight, setCenter]];
+}
+
+-(void) setProfileImageConstraints
+{
+    NSLayoutConstraint *fromCoverImage = [NSLayoutConstraint constraintWithItem: self.profileImage attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem: self.coverImage attribute:NSLayoutAttributeBottom multiplier:1.0 constant:20];
```
Comment (jmburges) [inline horizontal-profile/FISViewController.m]: those `20` guys? Move them into a constant at the top called like `PADDING` or something

File: horizontal-profile/FISViewController.m [hunk 7]
```diff
@@ -0,0 +1,195 @@
...
+
+-(void) setTextLabelConstraints
+{
+    NSLayoutConstraint *nameFromRightOfProfilePic = [NSLayoutConstraint constraintWithItem:self.nameLabel attribute:NSLayoutAttributeLeft relatedBy:NSLayoutRelationEqual toItem:self.profileImage attribute:NSLayoutAttributeRight multiplier:1.0 constant:20];
+
+    NSLayoutConstraint *classFromRightOfProfilePic = [NSLayoutConstraint constraintWithItem:self.classNameLabel attribute:NSLayoutAttributeLeft relatedBy:NSLayoutRelationEqual toItem:self.profileImage attribute:NSLayoutAttributeRight multiplier:1.0 constant:20];
+
+    NSLayoutConstraint *factFromRightOfProfilePic = [NSLayoutConstraint constraintWithItem:self.interestingFactLabel attribute:NSLayoutAttributeLeft relatedBy:NSLayoutRelationEqual toItem:self.profileImage attribute:NSLayoutAttributeRight multiplier:1.0 constant:20];
+    
+    NSLayoutConstraint *fromTopOfPic = [NSLayoutConstraint constraintWithItem:self.nameLabel attribute:NSLayoutAttributeTop relatedBy:NSLayoutRelationEqual toItem:self.profileImage attribute:NSLayoutAttributeTop multiplier:1.0 constant:6];
```
Comment (jmburges) [inline horizontal-profile/FISViewController.m]: `6` different padding. :sparkles: magical numbers!!! :sparkles:

File: horizontal-profile/FISViewController.m (hunk 6 above)
Comment (TOMalley104) [inline horizontal-profile/FISViewController.m]: good idea, doin' it

File: horizontal-profile/FISViewController.m (hunk 7 above)
Comment (TOMalley104) [inline horizontal-profile/FISViewController.m]: will be handled by the padding constant discussed in previous comment

**Comment by markedwardmurray** [comment]
solution branch doesn't need to be a PR

**Pull request closed without merging**